drivers/synopsys/ufs/dw_ufs.c

```diff
@@ -97,10 +97,21 @@ static int dwufs_phy_set_pwr_mode(ufs_params_t *params)
 	int result;
 	unsigned int data, tx_lanes, rx_lanes;
 	uintptr_t base;
+	unsigned int flags;
 
 	assert((params != NULL) && (params->reg_base != 0));
 
 	base = params->reg_base;
+	flags = params->flags;
+	if((flags & UFS_FLAGS_VENDOR_SKHYNIX) != 0) {
```

Comment:
MISRA C-2012 Rule 10.4 violation: The operands (flags&(U<<UFS_FLAGS_VENDOR_SKHYNIX)) and 0 have essential type categories unsigned 32-bit int and signed 8-bit int, which do not match.

To fix:

if ((flags & UFS_FLAGS_VENDOR_SKHYNIX) != 0U)

Also, this line has style check error:

drivers/synopsys/ufs/dw_ufs.c:106: ERROR: space required before the open parenthesis '('
	if((flags & UFS_FLAGS_VENDOR_SKHYNIX) != 0) {

Corrected in the suggested version.

Reply:
Thank you very much for pointing out these mistakes. I have fixed the errors.